src/components/document/tile-row.tsx

```diff
@@ -73,32 +75,42 @@ export class TileRowComponent extends BaseComponent<IProps, IState> {
 
   public render() {
     const { model } = this.props;
-    const { isSectionHeader, sectionId } = model;
+    const { isSectionHeader, sectionId, tiles } = model;
     // ignore height setting for section header rows
     const height = !isSectionHeader
                       ? this.props.height || model.height
                       : undefined;
     const style = height ? { height } : undefined;
+    const renderableTiles = tiles?.filter(tile => this.isTileRenderable(tile.tileId));
+    if (!isSectionHeader && !renderableTiles.length) return null;
     return (
       <div className={`tile-row`} data-row-id={model.id}
           style={style} ref={elt => this.tileRowDiv = elt}>
         { isSectionHeader && sectionId
           ? <SectionHeader type={sectionId}/>
-          : this.renderTiles(height)
+          : this.renderTiles(renderableTiles, height)
         }
         {!this.props.readOnly && this.renderDragDropHandles()}
       </div>
     );
   }
 
-  private renderTiles(rowHeight?: number) {
+  private isTileRenderable(tileId: string) {
+    const tile = this.props.tileMap.get(tileId);
+    return !!tile && (!tile.display || isShowingTeacherContent(this.stores));
+  }
+
+  private getTileWidth(tileId: string, tiles: TileLayoutModelType[]) {
+    // for now, distribute tiles evenly
+    return 100 / (tiles.length || 1);
```

Comment:
Is there a limit to the number of tiles in a row?  I added 5 in the JSON and it still looked ok.  Just curious if there is intended to be some tile limit in each row.

Reply:
No internal limit. Presumably, the user will stop adding tiles to the row when it becomes unusable for him/her. 😉